### PR TITLE
Delete quiz data

### PR DESCRIPTION
**Pull request by khaled-samy**
# Solves #47  
- add delete quiz route

### CODE REVIEW

File: __test__/teacher.test.ts [hunk 1]
```diff
@@ -21,6 +22,42 @@ const teacherToken = 'token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjE
...
+    const { body: { message } } = await supertest(app)
+      .delete('/api/v1/teacher/quiz/quiz-1')
+      .send(successDelete)
+      .expect(401)
+      .expect('Content-Type', /json/);
+
+    const actual = message;
+    const expected = 'Unauthorized';
+
+    expect(actual).toEqual(expected);
```
Comment (Mu7ammadAbed) [inline __test__/teacher.test.ts:58]:
```suggestion
    expect(message).toEqual('Unauthorized');
```

File: server/controllers/student/quizeDetails.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { NextFunction, Request, Response } from 'express';
+import { quizDetails } from '../../database/queries';
+import { CustomError } from '../../errors';
+
+export default async ({ params, cookies }:Request, res:Response, next:NextFunction) => {
```
Comment (Mu7ammadAbed) [inline server/controllers/student/quizeDetails.ts:5]: What is the purpose of this controller?
and why did you check for the token here? it's not the job of this controller to check if the user is authorized or not!

File: server/controllers/teacher/delete-quiz.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { NextFunction, Response } from 'express';
+import { deleteQuiz } from '../../database/queries';
+import { CustomError } from '../../errors';
+import { UserAuth } from '../../interfaces';
+
+export default async (req:UserAuth, res:Response, next:NextFunction) => {
+  const { params } = req;
+
+  try {
+    const { rowCount } = await deleteQuiz(params.quizId);
```
Comment (Mu7ammadAbed) [inline server/controllers/teacher/delete-quiz.ts:10]: You already destructured the params, complete your favor and destructure the quizId and then validate it

File: server/utils/models/deleted-data.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const successDelete = {
```
Comment (Mu7ammadAbed) [inline server/utils/models/deleted-data.ts:1]: Why do you need to use this in order to delete a quiz?

File: __test__/teacher.test.ts [hunk 5]
```diff
@@ -21,6 +22,42 @@ const teacherToken = 'token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjE
 beforeEach(dbBuild);
 afterAll(() => dbConnection.end());
 
+const cookie = 'token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjMsInJvbGUiOiJ0ZWFjaGVyIiwiaWF0IjoxNjUwNzg1Mzk3fQ.ErDrf4OSpcRWmrQACekbQBNwCRbBo1dmdBHRDio9b4w;';
+
+describe('DELETE /api/v1/teacher/quiz/:quizId', () => {
+  it('should return 200 and all quizzes teacher data as json response', async () => {
+    const res = await supertest(app)
+      .delete('/api/v1/teacher/quiz/quiz-1')
+      .send(successDelete)
```
Comment (Mu7ammadAbed) [inline __test__/teacher.test.ts:31]: You only need the `quizId` in order to delete a quiz so, there's no need to send this body, right?

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — DISMISSED** [review]

**Review by zaher-aa — DISMISSED** [review]

**Review by amjed-98 — APPROVED** [review]

File: server/controllers/teacher/delete-quiz.ts [hunk 6]
```diff
@@ -6,11 +6,12 @@ import { UserAuth } from '../../interfaces';
 
 export default async (req:UserAuth, res:Response, next:NextFunction) => {
   const { params: { quizId } } = req;
+  const { user: { userId: teacherId } } = req;
```
Comment (zaher-aa) [inline server/controllers/teacher/delete-quiz.ts:9]: This is nicer and cleaner
```ts
const { user: { userId: teacherId }, params: { quizId } } = req;
```

**Review by zaher-aa — APPROVED** [review]

File: __test__/teacher.test.ts [hunk 7]
```diff
@@ -215,3 +180,45 @@ describe('POST /api/v1/teacher/quiz', () => {
...
+      .set({ Cookie: teacherToken })
+      .expect(200);
+
+    expect(res.body.message).toBe('Success delete');
+  });
+
+  it('should return rowCount = 0', async () => {
+    const { rowCount } = await checkDeletedQuiz('quiz-1111111111111', 3);
+    expect(rowCount).toEqual(0);
+  });
```
Comment (Mu7ammadAbed) [inline __test__/teacher.test.ts:197]:
```suggestion
  it('Delete test quiz-1111111111111 for the second time to check if its still existing', async () => {
    const res = await supertest(app)
      .delete('/api/v1/teacher/quiz/quiz-1111111111111')
      .set({ Cookie: teacherToken })
      .expect(400);

    expect(res.body.message).toBe('No quiz to delete it');
  });
```

File: server/database/queries/teacher/check-deleted-quiz.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+import connections from '../../config/connections';
+
+export default (quizId: String, teacherId:Number) => connections
```
Comment (Mu7ammadAbed) [inline server/database/queries/teacher/check-deleted-quiz.ts:3]: There's no need for this query, we can check if the quiz has been deleted or not by trying to delete it again

**Review by Mu7ammadAbed — APPROVED** [review]

**Pull request merged**